components/feature/top-sites/src/main/java/mozilla/components/feature/top/sites/DefaultTopSitesStorage.kt

```diff
@@ -31,13 +30,10 @@ class DefaultTopSitesStorage(
     private val historyStorage: PlacesHistoryStorage,
     private val defaultTopSites: List<Pair<String, String>> = listOf(),
     coroutineContext: CoroutineContext = Dispatchers.IO
-) : TopSitesStorage, Observable<TopSitesStorage.Observer> by ObserverRegistry() {
+) : TopSitesMiddleware.Storage {
 
     private var scope = CoroutineScope(coroutineContext)
 
-    // Cache of the last retrieved top sites
-    var cachedTopSites = listOf<TopSite>()
-
     init {
         if (defaultTopSites.isNotEmpty()) {
```

Comment:
@pocmo Also, interested to know if this should belong in the Middleware. This functional block adds a list of default sites (Google, Wikipedia, Pocket) as part of first run and is only ran once.

Reply:
I do not have a strong opinion here. The advantage of the middleware is that it can potentially see (or wait for) more data to make a decision. For example it could wait for the `SetRegionAction` and then add the default top sites for this region.